hummingbot/strategy/pure_market_making/pure_market_making.pyx

```diff
@@ -148,8 +150,37 @@ cdef class PureMarketMakingStrategy(StrategyBase):
         self._last_own_trade_price = Decimal('nan')
         self._should_wait_order_cancel_confirmation = should_wait_order_cancel_confirmation
 
+        # deprecation warning for order level script user
+        self._last_order_level_warning_timestamp = 0
+        
         self.c_add_markets([market_info.market])
 
+    def initialize_order_level_spread(self, spread_list: List[Decimal]) -> List[Decimal]:
+        '''
+        convert to cumsum to remove the need to calculate each time it is used
+        if not enough element in list, append additional element based on last spread
+        '''
+
+        if spread_list is None:
```

Comment:
If the spread_list can be None, the type hint should be Optional[List[Decimal]]

Reply:
removed spread_list is None check as input should already be list of decimal or empty list